code/_onclick/hud/screen_objects/base_screen_objects.dm

```diff
@@ -1090,6 +1090,12 @@ obj/screen/fire/DEADelize()
 		if (G.active && G.overlay)//check here need if someone want call this func directly
 			overlays |= G.overlay
 
+	if(istype(H.back,/obj/item/weapon/rig))
+		var/obj/item/weapon/rig/O = H.back
+		if(O.visor && O.visor.active && O.visor.vision && O.visor.vision.glasses && (!O.helmet || (H.head && O.helmet == H.head)))
```

Comment:
jesus fuck thats a lot of checks.

Reply:
There's a lot of things i don't understand here. The first one is, why is this snippet necessary at all. Since there's a call to process_glasses in process_rig